### PR TITLE
Playlist Item Timeline

### PR DESCRIPTION
**Pull request by ezequielc**
## Description

- Add `timeline` property for Playlist Items
- Check if item is scheduled to play before playing item. Code is duplicated from https://github.com/Rise-Vision/viewer/blob/master/src-js/Viewer/Parser/TimelineParser.js
- Sends Done if no item is scheduled to play. 

## Motivation and Context
Playlist Improvements

## How Has This Been Tested?
Locally and with Charles proxy

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Comment by ezequielc** [comment]
@alex-deaconu This is the base PR for playlist item Timeline, more changes and improvements will be added as additional PRs. Please review. Thanks

**Review by alex-deaconu — APPROVED** [review]

**Comment by alex-deaconu** [comment]
@stulees @olegrise this is the second PR, for the main Timeline implementation.

To confirm, the main features is the addition of a `timeline` attribute, which contains an object defining the timeline settings. These can be configured in Apps at:
https://apps-stage-1.risevision.com.

The timeline parsing (`timeline.js`) is existing functionality from the Viewer. We have disabled (commented out for now) checks for Preview and the `applyTimeline` url parameter. If time permits we will address this, otherwise it will be a future improvement.

Also, while we are working on a few extra unit tests, the main functionality is reviewed and tested at this point.

Please review. Thanks!

File: src/timeline.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  }
+  // if (!factory.applies()) {
+  //   return true;
+  // }
+
+  if (!d) {
+    d = new Date();
+  }
+
+  if (item.useDisplayTime) {
```
Comment (stulees) [inline src/timeline.js:118]: @alex-deaconu Is this block necessary to keep if these lines are commented out? Thanks

File: package.json [hunk 2]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "rise-playlist",
-  "version": "1.0.7",
+  "version": "1.0.8",
```
Comment (stulees) [inline package.json:3]: @alex-deaconu This is minor, but this is a feature addition so maybe is more appropriate to bump to `1.1.0` ?

**Review by stulees — APPROVED** [review]
@alex-deaconu Some minor questions, otherwise looks good to me, thanks

File: src/timeline.js (hunk 1 above)
Comment (ezequielc) [reply src/timeline.js:118]: @stulees This code was commented out in the original Viewer code. I am good with removing it if you think it is better. Thanks
@alex-deaconu fyi

File: src/timeline.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  if (item.recurrenceType === RECURRENCE_TYPE.DAILY) {
+    const days = _diffDays(startDate, d);
+
+    if (parseInt(days % item.recurrenceFrequency) != 0) {
+      return false;
+    }
+  } else if (item.recurrenceType === RECURRENCE_TYPE.WEEKLY) {
+    const weeks = _diffWeeks(startDate, d);
+    // unit test the diffWeeks function
+//			String utres ='';
```
[the remark this answers is not in the document]
Comment (ezequielc) [reply src/timeline.js:172]: @stulees Same as the above, this was commented out in the original Viewer code. Thanks
@alex-deaconu fyi

File: package.json (hunk 2 above)
Comment (alex-deaconu) [reply package.json:3]: @stulees I will do a major bump as suggested. Thanks!

File: src/timeline.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+};
+
+const _diffWeeks = (fromDate, toDate) => {
+  const days = _diffDays(fromDate, toDate);
+  const weeks = (days + fromDate.getDay()) / 7;
+
+  return parseInt(weeks);
+};
+
+// factory.applies = () => {
```
[the remark this answers is not in the document]
Comment (ezequielc) [reply src/timeline.js:96]: @stulees This was temporarily commented out as not needed in the current implementation, but we have a Nice-to-have task that will use it (skip timeline check, as used in Schedule details page). Alex is actually pursuing it, so we will change this soon. Thanks
@alex-deaconu fyi

File: package.json (hunk 2 above)
Comment (ezequielc) [reply package.json:3]: @stulees Good point, I will update based on your suggestion. Thanks!

**Comment by alex-deaconu** [comment]
@stulees I've pushed a commit to bump the version as suggested. As for the commented out code, it is addressed in https://github.com/Rise-Vision/rise-playlist-new/pull/74. Thanks.

**Pull request merged**